apps/condo/domains/property/components/panels/Builder/BuildingPanelEdit.tsx

```diff
@@ -218,6 +219,8 @@ export const BuildingPanelEdit: React.FC<IBuildingPanelEditProps> = (props) => {
     const sections = mapEdit.sections
     const address = get(property, 'address')
 
+    const [duplicatedUnitIds, setDuplicatedUnitIds] = useState<string[]>([])
```

Comment:
Why you need a state to handle duplicates? That's too heavy solution, that could reduce performance.
Is it possible to do this only with special method of `MapConstructor`?

Reply:
The state is needed so that after validation, duplicates can be highlighted. I didn't see any major changes in performance.